Dima/netflix-roulette/src/Pages/Home/Home.tsx

```diff
@@ -30,84 +30,80 @@ const connector = connect(mapStateToProps, mapDispatchToProps)
 
 type PropsFromRouteAndRedux = ConnectedProps<typeof connector> & RouteComponentProps
 
-class Home extends React.Component<PropsFromRouteAndRedux> {
+function Home(props: PropsFromRouteAndRedux) {
 
-  componentDidMount() {
-    const URLData = parse(this.props.location.search) as { search: string, searchBy: string };
+  useEffect(() => {
```

Comment:
you don't need several useEffects here. you need to make an api call when location is changed. The useEffect on the 40 line will be called on mount too!!!